app/models/version.rb

```diff
@@ -11,14 +18,24 @@ def initialize(version_string)
           commit: match[3])
   end
 
+  def self.production?
+    Rails.env.production?
+  end
+
+  def self.read_version_file
+    File.read(Rails.root.join('VERSION')).strip
+  rescue Errno::ENOENT => _
+    ""
+  end
+
+  def self.read_version_from_git
+    `git describe --long --tags`.strip
+  end
+
   def self.load_version
-    if Rails.env == 'production'
-      File.read(Rails.root.join('VERSION')).strip
-    else
-      git_version = `git describe --long --tags`.strip
-      raise 'Could not determine backend version via git.' if git_version.empty?
-      git_version
-    end
+    version = production? ? read_version_file : read_version_from_git
+    raise CouldNotDetermineVersion if version.empty?
+    version
   end
```

Comment:
If you put the class methods into a `class << self` block, you can make all the methods except for `load_version` private.